pkgs/development/python-modules/pyxdg/default.nix

```diff
@@ -6,24 +6,16 @@
 
 buildPythonPackage rec {
   pname = "pyxdg";
-  version = "0.26";
+  version = "0.27";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "fe2928d3f532ed32b39c32a482b54136fe766d19936afc96c8f00645f9da1a06";
+    sha256 = "19f5j5mxp7ff0vp33s32qbpdi65iiwha0bj641gl70pdwnm97gc0";
   };
 
   # error: invalid command 'test'
   doCheck = false;
```

Comment:
We can easily get the tests running, they're included in the git checkout. But there are four failing ones. Can you look into those?

```diff
-  # error: invalid command 'test'
-  doCheck = false;
+  checkInputs = [
+    nose
+  ];
+  checkPhase = ''
+    nosetests
+  '';
```

Reply:
I've resolved the issues, but I'm not going to be able to spend any time looking into test failures.

Reply:
```
Finished executing pythonRemoveTestsDir
running install tests
.........................................EF..FFEEEF.....
======================================================================
ERROR: test_by_name (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 135, in test_by_name
    self.check_mimetype(dot_c, 'text', 'x-csrc')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AttributeError: 'NoneType' object has no attribute 'media'

======================================================================
ERROR: test_get_type_by_contents (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 51, in test_get_type_by_contents
    self.check_mimetype(imgpng, 'image', 'png')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AttributeError: 'NoneType' object has no attribute 'media'

======================================================================
ERROR: test_get_type_by_data (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 37, in test_get_type_by_data
    self.check_mimetype(imgpng, 'image', 'png')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AttributeError: 'NoneType' object has no attribute 'media'

======================================================================
ERROR: test_get_type_by_name (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 33, in test_get_type_by_name
    self.check_mimetype(appzip, 'application', 'zip')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AttributeError: 'NoneType' object has no attribute 'media'

======================================================================
FAIL: test_canonical (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 146, in test_canonical
    self.check_mimetype(text_xml.canonical(), 'application', 'xml')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AssertionError: 'text' != 'application'
- text
+ application


======================================================================
FAIL: test_get_type (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 59, in test_get_type
    self.check_mimetype(imgpng, 'image', 'gif')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AssertionError: 'text' != 'image'
- text
+ image


======================================================================
FAIL: test_get_type2 (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 83, in test_get_type2
    self.check_mimetype(Mime.get_type2(example_file('test.gif')), 'image', 'gif')
  File "/build/source/test/test-mime.py", line 14, in check_mimetype
    self.assertEqual(mimetype.media, media)
AssertionError: 'application' != 'image'
- application
+ image


======================================================================
FAIL: test_inheritance (test-mime.MimeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/test/test-mime.py", line 157, in test_inheritance
    self.assertEqual(text_python.inherits_from(), set([text_plain, app_executable]))
AssertionError: Items in the second set but not the first:
MIMEtype('text', 'plain')
MIMEtype('application', 'x-executable')

----------------------------------------------------------------------
Ran 56 tests in 3.407s
```

Reply:
Yeah, I've seen those and couldn't immediately repro them in a python shell. I'll take another stab tomorrow.

Reply:
Can you please add back a comment why those tests are disabled?